### ISSUE TITLE
Support yielding over pcall from C API (lua_pcallk)

### ISSUE THREAD

**Issue by daurnimator** [LuaJIT/LuaJIT#48]
LuaJIT currently supports yielding over pcall from lua code:

```
LuaJIT 2.0.4 -- Copyright (C) 2005-2015 Mike Pall. http://luajit.org/
JIT: ON CMOV SSE2 SSE3 SSE4.1 fold cse dce fwd dse narrow loop abc sink fuse
> local co = coroutine.create(function() print("A"); print(pcall(function() coroutine.yield() print("B") end)); print("C"); end); coroutine.resume(co); coroutine.resume(co);
A
B
true
C
```

But LuaJIT does not have a way to perform the above from C code.

Lua 5.2 and 5.3 extend `lua_pcall` via `lua_pcallk` which takes an additional 'continuation' argument.

**Comment by zhaozg** [comment]
Priority +1

**Comment by ThePhD** [comment]
I recently ran into this while working on the Lua 5.2 <-> luajit compatbility shim for my code. Would appreciate this being implemented for v2.1.x!
